test/AspNet.Security.OAuth.Providers.Tests/VkId/VkIdTests.cs

```diff
@@ -0,0 +1,55 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using Microsoft.AspNetCore.DataProtection;
+using NSubstitute;
+
+namespace AspNet.Security.OAuth.VkId;
+
+public class VkIdTests : OAuthTests<VkIdAuthenticationOptions>
+{
+    public VkIdTests(ITestOutputHelper outputHelper)
+        : base(outputHelper)
+    {
+        LoopbackRedirectHandler.LoopbackParameters.Add("device_id", "1111");
+        LoopbackRedirectHandler.LoopbackParameters.Add("type", "code_v2");
+    }
+
+    public override string DefaultScheme => VkIdAuthenticationDefaults.AuthenticationScheme;
+
+    protected internal override void RegisterAuthentication(AuthenticationBuilder builder)
+    {
+        var dataProtector = (IDataProtector)DataProtectionProvider.Create("test");
+        var fakeDataProtector = Substitute.For<IDataProtector>();
+
+        fakeDataProtector.Protect(Arg.Any<byte[]>())
+            .Returns(x => dataProtector.Protect(x.Arg<byte[]>()));
+        fakeDataProtector.Unprotect(Arg.Is<byte[]>(x => !x.SequenceEqual(Array.Empty<byte>())))
+            .Returns(x => dataProtector.Unprotect(x.Arg<byte[]>()));
+
+        // Use fake DP with empty AuthenticationProperties for ExchangeCodeAsync state validation
+        fakeDataProtector.Unprotect(Arg.Is<byte[]>(x => x.SequenceEqual(Array.Empty<byte>())))
+            .Returns([1, 0, 0, 0, 0, 0, 0, 0]);
```

Comment:
Could you make this use a slightly more realistic value that matches a non-empty `scope` value in the payload?